### DISCUSSION TITLE
Error message about PARPROF_PFT_DIR_Z

### DISCUSSION THREAD

**Discussion by jennykowalcz** [NGEET/fates#1274] [Q&A] [answered]
Hi all! I'm looking for help understanding an error message I have not encountered before. My FATES run failed after ~195 years with:
<img width="835" alt="Screenshot 2024-10-29 at 1 35 54 PM" src="https://github.com/user-attachments/assets/2defdd69-31b5-4fce-8e3e-372768097673">

I rebuilt with DEBUG=TRUE and now it reads:
<img width="899" alt="Screenshot 2024-10-30 at 10 19 41 AM" src="https://github.com/user-attachments/assets/f3335dd8-946e-44ce-8aa0-69bd33d00a45">

This is a full-complexity FATES run over tropical S. America with two evergreen tropical tree PFTs representing early and late successionals. I'm using two-stream radiation and have nclmax set to 3.

**Comment by jennykowalcz** [comment]
It looks like the third subscript of this array corresponds to `nlevleaf`

**Reply by ckoven** [reply]
It looks like it might plausibly be caused by the underlying problem that https://github.com/NGEET/fates/pull/1269 tries to solve? Maybe try pulling that branch or cherry-picking the commit into your branch and see if that fixes it?

**Reply by jennykowalcz** [reply]
Aha, okay, thanks, I'll try that and report back!

**Reply by jennykowalcz** [reply]
Darn, sadly that didn't fix it. It still crashes at the same point with the same error after cherry-picking [the commit](https://github.com/NGEET/fates/pull/1269/commits/26429c66cf223dbf68db41cc244c4c58014fec3a) into my branch.

Digging into the code, I see PARPROF_PFT_DIR_Z is allocated as parprof_pft_dir_z(ncan,numpft,nveg) where nveg = maxval(this%ncan(:,:)) . (In the main branch ncan is changed to nleaf with [this commit](https://github.com/NGEET/fates/commit/a2b3eada362248de87116f93da31c7337c700ce7) ).

@rgknox does anything jump out at you here?

I'm going to try variations of this case with Norman radiation and also with the main branch..

**Reply by jennykowalcz** [reply]
If I switch to Norman radiation and continue from the last restart at 190 years, it gets past the crash point and completes the 10 year block successfully. Next I will try with two-stream radiation but with the main branch.

**Comment by jennykowalcz** [comment, marked as answer]
Moved to [issue 1279](https://github.com/NGEET/fates/issues/1279)
